versions.yaml

```diff
@@ -0,0 +1,34 @@
+# This file defines all the Istio versions supported by this operator.
+# The latest version must be listed first, so that the CRDs from the latest version
+# are copied to the config/crd directory.
+latest:
+  repo: https://github.com/istio/istio
+  branch: master
+  commit: 62a9db5971f3bd0fb0f5e8872e864101446a3c5e
+v1.19.3:
+  repo: https://github.com/istio/istio
+  commit: 41218ffa66c12acf77ad08b3ef07d296218d0929
+  charts:
+    - https://istio-release.storage.googleapis.com/charts/base-1.19.3.tgz
+    - https://istio-release.storage.googleapis.com/charts/istiod-1.19.3.tgz
+    - https://istio-release.storage.googleapis.com/charts/gateway-1.19.3.tgz
+    - https://istio-release.storage.googleapis.com/charts/cni-1.19.3.tgz
+    - https://istio-release.storage.googleapis.com/charts/ztunnel-1.19.3.tgz
+v1.19.1:
```

Comment:
Any reason to have two 1.19 listed here?

Reply:
I'm going to need two patch releases of the same minor release when I implement control plane upgrades. Also, in the end, the operator will support multiple patch versions like this, so we might as well add some now.

Reply:
What's the plan as new versions are released? Is this list going to grow forever? Or we will keep a fixed number of releases? This file can be edited by the user for any purpose, right?

Will there be any distinction between supported and non-supported versions in this file?

Reply:
> What's the plan as new versions are released? Is this list going to grow forever? Or we will keep a fixed number of releases? 
This is TBD. We can decide which versions to include in the operator every time we do a release. 

For product releases, we'll probably include the versions from the previous operator release + new patch releases for each minor version. We can drop versions from the operator just like we do in 2.x.

For the upstream operator, we can include all released Istio versions that are still supported by the community.

In any case, the decision on which versions should be included in the operator is not related to this PR. This PR merely allows us to include multiple versions in the operator.

> This file can be edited by the user for any purpose, right?
Not by the user, but by operator developers. But, yes, anyone that's building the operator can add whatever versions they want by editing this file and running `make gen`. 

> Will there be any distinction between supported and non-supported versions in this file?
There's currently no need to make a distinction. The operator allows users to deploy any version that's specified here (well, any version that's in the `resources/` dir). If needed, we can add the ability to mark supported vs unsupported versions later.